app/src/main/java/com/gophillygo/app/utils/UserUuidUtils.java

```diff
@@ -0,0 +1,51 @@
+package com.gophillygo.app.utils;
+
+import android.content.Context;
+import android.content.SharedPreferences;
+import android.util.Log;
+
+import com.gophillygo.app.R;
+
+import java.util.UUID;
+
+public class UserUuidUtils {
+
+    private static final String LOG_LABEL = "UserUUidUtils";
+    private final Context context;
+
+    public UserUuidUtils(Context context) {
+        this.context = context;
+    }
+
+    /**
+     * Get the randomly generated ID for this app install from Shared Preferences, or create and
+     * put one there if it does not yet exist, which can happen if this is the first app install,
+     * or if the shared preferences file was erased (i.e., by the user clearing app settings.)
+     *
+     * @return String representation of a UUID
+     */
+    public String getUserUuid() {
```

Comment:
Given that the user pattern of this class everywhere it is used is:
```java
UserUuidUtils uuidUtils = new UserUuidUtils(context);
uuid = uuidUtils.getUserUuid();
```
Consider making this method static and adding a `Context` parameter, to avoid needing to instantiate the class.